service/storage_proxy.cc

```diff
@@ -2155,36 +2156,45 @@ future<result<>> storage_proxy::mutate_begin(unique_response_handler_vector ids,
 
 // this function should be called with a future that holds result of mutation attempt (usually
 // future returned by mutate_begin()). The future should be ready when function is called.
-future<> storage_proxy::mutate_end(future<result<>> mutate_result, utils::latency_counter lc, write_stats& stats, tracing::trace_state_ptr trace_state) {
+future<result<>> storage_proxy::mutate_end(future<result<>> mutate_result, utils::latency_counter lc, write_stats& stats, tracing::trace_state_ptr trace_state) {
     assert(mutate_result.available());
     stats.write.mark(lc.stop().latency());
     if (lc.is_start()) {
         stats.estimated_write.add(lc.latency());
     }
+    auto exception_handler = overloaded_functor {
+        [&] (const mutation_write_timeout_exception& ex) {
+            // timeout
+            tracing::trace(trace_state, "Mutation failed: write timeout; received {:d} of {:d} required replies", ex.received, ex.block_for);
+            slogger.debug("Write timeout; received {} of {} required replies", ex.received, ex.block_for);
+            stats.write_timeouts.mark();
+        }
+    };
     try {
-        mutate_result.get().value(); // .value() throws the error, if there is any
+        auto res = mutate_result.get();
+        if (res.has_error()) {
```

Comment:
Can you just write `!res` here?

Reply:
Yes, it appears that the result type has a bool operator which works in this way. I'll change invocations of `has_error` to usages of the bool operator.